analyze_model.py

```diff
@@ -53,11 +53,17 @@ def _rernd_layers(model, layers_indices):
 
     @staticmethod
     def total_edges(model):
+        return AnalyzeModel.total_edges_in_layers(model, list(range(len(model.layers))))
+
+    @staticmethod
+    def total_edges_in_layers(model, layers=[], count_nonzero_only=False):
         edges = 0
-        for layer in model.layers:
-            weights = layer.get_weights()
+        for layer in layers:
+            weights = model.layers[layer].get_weights()
             if weights:
                 edges += weights[0].size
+                if count_nonzero_only:
+                    edges -= np.count_nonzero(weights[0])
```

Comment:
I think you have a bug here? if you are only counting non-zeroes, you should remove the zeros, not the non-zeroes, no?

Reply:
Looks like a bug, yes... wonder why the results look sane with this bug inside

Reply:
Fixed